commit cd945e14a332e63e395dc1df4fcacd09a5e7e892
Author: cazfi <cazfi@a0f10bec-cc02-0410-94fc-a9cfff90b4cd>
Date:   Sun May 15 07:16:05 2016 +0000

    When borders change owners and border vision is controlled by effects, adjust
    vision of the one losing the borders based on his/her own effect value.
    
    See bug #24313
    
    
    git-svn-id: svn://svn.gna.org/svn/freeciv/trunk@32714 a0f10bec-cc02-0410-94fc-a9cfff90b4cd

diff --git a/server/maphand.c b/server/maphand.c
index a2b36e2..822cee8 100644
--- a/server/maphand.c
+++ b/server/maphand.c
@@ -1913,21 +1913,22 @@ static void map_claim_border_ownership(struct tile *ptile,
 {
   struct player *ploser = tile_owner(ptile);
 
-  if (BORDERS_SEE_INSIDE == game.info.borders
-      || BORDERS_EXPAND == game.info.borders
-      || (powner && powner->server.border_vision)) {
-    if (ploser != powner) {
-      if (ploser) {
-        const v_radius_t radius_sq = V_RADIUS(-1, 0);
-
-        shared_vision_change_seen(ploser, ptile, radius_sq, FALSE);
-      }
-      if (powner) {
-        const v_radius_t radius_sq = V_RADIUS(1, 0);
+  if ((ploser != powner && ploser != NULL)
+      && (BORDERS_SEE_INSIDE == game.info.borders
+          || BORDERS_EXPAND == game.info.borders
+          || ploser->server.border_vision)) {
+    const v_radius_t radius_sq = V_RADIUS(-1, 0);
 
-        shared_vision_change_seen(powner, ptile, radius_sq, TRUE);
-      }
-    }
+    shared_vision_change_seen(ploser, ptile, radius_sq, FALSE);
+  }
+
+  if (powner != NULL
+      && (BORDERS_SEE_INSIDE == game.info.borders
+          || BORDERS_EXPAND == game.info.borders
+          || powner->server.border_vision)) {
+    const v_radius_t radius_sq = V_RADIUS(1, 0);
+
+    shared_vision_change_seen(powner, ptile, radius_sq, TRUE);
   }
 
   tile_set_owner(ptile, powner, psource);
